frontend/src/concepts/pipelines/content/compareRuns/metricsSection/markdown/MarkdownCompare.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import {
+  Bullseye,
+  Divider,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateHeader,
+  EmptyStateVariant,
+  Flex,
+  FlexItem,
+  Spinner,
+} from '@patternfly/react-core';
+
+import { PipelineRunKFv2 } from '~/concepts/pipelines/kfTypes';
+
+import { RunArtifact } from '~/concepts/pipelines/apiHooks/mlmd/types';
+import { FullArtifactPath } from '~/concepts/pipelines/content/compareRuns/metricsSection/types';
+import {
+  getFullArtifactPaths,
+  getFullArtifactPathLabel,
+} from '~/concepts/pipelines/content/compareRuns/metricsSection/utils';
+import { CompareRunsEmptyState } from '~/concepts/pipelines/content/compareRuns/CompareRunsEmptyState';
+import { MarkdownSelect } from './MarkdownSelect';
+
+type MarkdownCompareProps = {
+  runArtifacts?: RunArtifact[];
+  isLoaded: boolean;
+};
+
+export type MarkdownAndTitle = {
+  title: string;
+  markdown: string;
+};
+
+const MarkdownCompare: React.FC<MarkdownCompareProps> = ({ runArtifacts, isLoaded }) => {
+  const [expandedGraph, setExpandedGraph] = React.useState<MarkdownAndTitle | undefined>(undefined);
+
+  const fullArtifactPaths: FullArtifactPath[] = React.useMemo(() => {
+    if (!runArtifacts) {
+      return [];
+    }
+
+    return getFullArtifactPaths(runArtifacts);
+  }, [runArtifacts]);
+
+  const { configMap, runMap } = React.useMemo(() => {
+    const configMapBuilder: Record<string, MarkdownAndTitle[]> = {};
+    const runMapBuilder: Record<string, PipelineRunKFv2> = {};
+
+    fullArtifactPaths
+      .map((fullPath) => ({
+        run: fullPath.run,
+        title: getFullArtifactPathLabel(fullPath),
+        uri: fullPath.linkedArtifact.artifact.getUri(),
+      }))
+      .filter((markdown) => !!markdown.uri)
+      .forEach(async ({ uri, title, run }) => {
+        const data = uri; // TODO: fetch data from uri
```

Comment:
nit: Is there a JIRA associated with this TODO?

Reply:
not yet, im working on the spike or the data fetching

Reply:
added